client/cmd/job/change_namespace.go

```diff
@@ -118,25 +120,272 @@ func (c *changeNamespaceCommand) sendChangeNamespaceRequest(jobName string) erro
 	return err
 }
 
+func getJobParentsFilePathList(fileFS afero.Fs, jobDirPath string) ([]string, error) {
+	var parentsFilePathList []string
+	var errorMsg string
+	splitDirPaths := strings.Split(jobDirPath, "/")
+	for i := range splitDirPaths {
+		pathNearSpecIdx := len(splitDirPaths) - i
+		rootToNearSpecPaths := splitDirPaths[:pathNearSpecIdx]
+		parentDirPath := filepath.Join(rootToNearSpecPaths...)
+		parenFilePath := filepath.Join(parentDirPath, "this.yaml")
+		exist, err := afero.Exists(fileFS, parenFilePath)
+		if err != nil {
+			errorMsg += fmt.Sprintf("Unable to read directory %s to check 'this.yaml', err: %s\n", parenFilePath, err.Error())
+			continue
+		}
+		if exist {
+			parentsFilePathList = append(parentsFilePathList, parenFilePath)
+		}
+	}
+	var err error
+	if len(errorMsg) > 0 {
+		err = errors.New(errorMsg)
+	}
+	return parentsFilePathList, err
+}
+
+func mergeYAMLFiles(fs afero.Fs, sourcePath, targetPath string) error {
+	sourceObj, err := readYaml(fs, sourcePath)
+	if err != nil {
+		return err
+	}
+	targetObj, err := readYaml(fs, targetPath)
+	if err != nil {
+		return err
+	}
+	mergedObj, err := mergeObjects(sourceObj, targetObj)
+	if err != nil {
+		return err
+	}
+	mergedFile, err := yaml.Marshal(mergedObj)
+	if err != nil {
+		return err
+	}
+	return afero.WriteFile(fs, targetPath, mergedFile, os.ModePerm)
+}
+
+func mergeObjects(interfaces ...interface{}) (interface{}, error) {
+	merged := make(map[string]interface{})
+	for _, intf := range interfaces {
+		val := reflect.ValueOf(intf)
+		if val.Kind() != reflect.Map {
+			return nil, errors.New("automatic merge not possible")
+		}
+		iter := val.MapRange()
+		for iter.Next() {
+			key := iter.Key().Interface()
+			value := iter.Value().Interface()
+			if existingValue, exists := merged[key.(string)]; exists {
+				mergedValue := mergeValues(existingValue, value)
+				merged[key.(string)] = mergedValue
+			} else {
+				merged[key.(string)] = value
+			}
+		}
+	}
+	return merged, nil
+}
+
+func mergeValues(value1, value2 interface{}) interface{} {
+	val1 := reflect.ValueOf(value1)
+	val2 := reflect.ValueOf(value2)
+
+	if val1.Kind() != val2.Kind() {
+		return value2
+	}
+
+	switch val1.Kind() {
+	case reflect.Map:
+		merged := make(map[string]interface{})
+		iter1 := val1.MapRange()
+		for iter1.Next() {
+			key := iter1.Key().Interface()
+			value := iter1.Value().Interface()
+			merged[key.(string)] = value
+		}
+		iter2 := val2.MapRange()
+		for iter2.Next() {
+			key := iter2.Key().Interface()
+			value := iter2.Value().Interface()
+			if existingValue, exists := merged[key.(string)]; exists {
+				mergedValue := mergeValues(existingValue, value)
+				merged[key.(string)] = mergedValue
+			} else {
+				merged[key.(string)] = value
+			}
+		}
+		return merged
+
+	default:
+		return value2
+	}
+}
+
+func findConflicts(incomming, existing interface{}, incomingFileName, existingFileName, path string) []string {
```

Comment:
it would better to find conflict without convert to the object first (if possible). Or we could use 3rd party lib